pkg/nimble/netif/nimble_netif.c

```diff
@@ -0,0 +1,621 @@
+
+/* allocation of memory for buffering IP packets when handing them to NimBLE */
+static os_membuf_t _mem[OS_MEMPOOL_SIZE(MBUF_CNT, MBUF_SIZE)];
+static struct os_mempool _mem_pool;
+static struct os_mbuf_pool _mbuf_pool;
+
+/* notify the user about state changes for a connection context */
+static void _notify(int handle, nimble_netif_event_t event)
+{
+    if (_eventcb) {
+        _eventcb(handle, event);
+    }
+}
+
+static void _netif_init(gnrc_netif_t *netif)
+{
+    (void)netif;
+
+    /* save the threads context pointer, so we can set its flags */
+    _netif_thread = (thread_t *)thread_get(thread_getpid());
+
+#ifdef MODULE_GNRC_SIXLOWPAN
+    /* we disable fragmentation for this device, as the L2CAP layer takes care
+     * of this */
+    _nimble_netif->sixlo.max_frag_size = 0;
+#endif
+}
+
+static int _send_pkt(nimble_netif_conn_t *conn, gnrc_pktsnip_t *pkt)
+{
+    int res;
+    int num_bytes = 0;
+
+    if (conn->coc == NULL) {
+        return NIMBLE_NETIF_DEVERR;
+    }
+
+    /* copy the data into a newly allocated mbuf */
+    struct os_mbuf *sdu = os_mbuf_get_pkthdr(&_mbuf_pool, 0);
+    if (sdu == NULL) {
+        return NIMBLE_NETIF_NOMEM;
+    }
+    while (pkt) {
+        res = os_mbuf_append(sdu, pkt->data, pkt->size);
+        if (res != 0) {
+            os_mbuf_free_chain(sdu);
+            return NIMBLE_NETIF_NOMEM;
+        }
+        num_bytes += (int)pkt->size;
+        pkt = pkt->next;
+    }
+
+    /* send packet via the given L2CAP COC */
+    do {
+        res = ble_l2cap_send(conn->coc, sdu);
+        if (res == BLE_HS_EBUSY) {
+            thread_flags_wait_all(FLAG_TX_UNSTALLED);
+        }
+    } while (res == BLE_HS_EBUSY);
+
+    if ((res != 0) && (res != BLE_HS_ESTALLED)) {
+        os_mbuf_free_chain(sdu);
+        return NIMBLE_NETIF_DEVERR;
+    }
+
+    return num_bytes;
+}
+
+static int _netif_send_iter(nimble_netif_conn_t *conn,
+                            int handle, void *arg)
+{
+    (void)handle;
+    _send_pkt(conn, (gnrc_pktsnip_t *)arg);
+    return 0;
+}
+
+static int _netif_send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt)
+{
+    (void)netif;
+    int res;
+
+    if (pkt->type != GNRC_NETTYPE_NETIF) {
+        return NIMBLE_NETIF_DEVERR;
+    }
+
+    gnrc_netif_hdr_t *hdr = (gnrc_netif_hdr_t *)pkt->data;
+    /* if packet is bcast or mcast, we send it to every connected node */
+    if (hdr->flags &
+        (GNRC_NETIF_HDR_FLAGS_BROADCAST | GNRC_NETIF_HDR_FLAGS_MULTICAST)) {
+        nimble_netif_conn_foreach(NIMBLE_NETIF_L2CAP_CONNECTED,
+                                  _netif_send_iter, pkt->next);
+        res = (int)gnrc_pkt_len(pkt->next);
+    }
+    /* send unicast */
+    else {
+        int handle = nimble_netif_conn_get_by_addr(
+            gnrc_netif_hdr_get_dst_addr(hdr));
+        nimble_netif_conn_t *conn = nimble_netif_conn_get(handle);
+        if (conn) {
+            res = _send_pkt(conn, pkt->next);
+        }
+        else {
+            res = NIMBLE_NETIF_NOTCONN;
```

Comment:
This should be or map to an `errno` (e.g. `ENOTCONN`) as per documentation of [the `send` operation](https://doc.riot-os.org/structgnrc__netif__ops.html#a86afb0a1108b6bbacf281c4aba13824c).

Reply:
fixed.